src/main/java/org/fogbowcloud/manager/core/threads/AttendSpawningOrdersThread.java

```diff
@@ -0,0 +1,129 @@
+		this.sshConnectivity = SshConnectivityUtil.getInstance();
+	}
+	
+	@Override
+	public void run() {
+		while(true) {
+			Order order = null;
+			try {
+				order = SharedOrderHolders.getInstance().getSpawningOrdersList().getNext();
+				if (order != null) {
+					this.processSpawningOrder(order);
+				} else {
+					LOGGER.info("There is no spawning order to be processed, sleeping attend spawning orders thread.");
+					Thread.sleep(this.sleepTime);
+				}
+			} catch (Throwable e) {
+				String orderId = order != null ? order.getId() : null;
+				LOGGER.error("Error while trying to thread sleep attending the order: [" + orderId + "]", e);
+			}
+		}
+	}
+
+	protected void processSpawningOrder(Order order) {
+		synchronized (order) {
+			OrderState orderState = order.getOrderState();
+			if (orderState.equals(OrderState.SPAWNING)) {
+				LOGGER.info("Trying to get an instance for order [" + order.getId() + "]");	
+				try {
+					OrderInstance orderInstance = order.getOrderInstance();
+					if (order.getType().equals(OrderType.COMPUTE)) {
+						processeComputeInstance(order, orderInstance);
+					}					
+				} catch (Exception e) {
+					LOGGER.error("Error while trying to get an instance for order: " + System.lineSeparator() + order, e);
+				}
+			} else {
+				LOGGER.info("This order state is not spawning for order [" + order.getId() + "]");
+			}
+		}
+	}
+
+	private void processeComputeInstance(Order order, OrderInstance orderInstance) throws OrderStateTransitionException {
+		synchronized (order) {
+			ComputeOrderInstance computeOrderInstance = null;
+			try {
+				computeOrderInstance = this.computePlugin.getInstance(order.getLocalToken(), orderInstance.getId());
+				if (computeOrderInstance.getState().equals(InstanceState.FAILED)) {
+					OrderStateTransitioner.transition(order, OrderState.FAILED);
+				} else {
+					if (computeOrderInstance.getState().equals(InstanceState.ACTIVE)) {
+						LOGGER.info("Processing active compute instance for order [" + order.getId() + "]");
+						this.setTunnelingServiceAddresses(order, computeOrderInstance);
+						if (isActiveConnectionFromInstance(computeOrderInstance)) {
+							OrderStateTransitioner.transition(order, OrderState.FULFILLED);
+						} else {
+							LOGGER.warn("Failed attempt to communicate with ssh connectivity for order [" + order.getId() + "]");
+						}
+					} else {
+						LOGGER.info("The compute instance is inactive for order [" + order.getId() + "]");
+					}
+				}			
+			} catch (Exception e) {
+				LOGGER.error("Error while trying to get the compute instance for order: " + System.lineSeparator() + order, e);
+			}
+		}
+		
+	}
+
+	private void setTunnelingServiceAddresses(Order order, ComputeOrderInstance computeOrderInstance) {
+		synchronized (order) {
+			try {
+				Map<String, String> externalServiceAddresses = tunnelingService.getExternalServiceAddresses(order.getId());
+				if (externalServiceAddresses != null) {
```

Comment:
Apparently your JUnit tests for this class do not cover the case that the externalServiceAddresses is null.